dss/stepfunctions/visitation/storage.py

```diff
@@ -28,38 +31,44 @@ class StorageVisitation(Visitation):
 
     state_spec = dict(replicas=None,
                       buckets=None,
+                      folder=None,
+                      quick=None,
                       work_result=dict(present=None,
                                        missing=None))
 
     prefix_chars = list(set(string.hexdigits.lower()))
 
     def job_initialize(self):
+        folders = StorageFolder.cls_by_name.keys()
+        require(self.folder in folders, f"Folder {self.folder} is not one of {folders}")
         if self._number_of_workers <= 16:
             self.work_ids = self.prefix_chars
         else:
             self.work_ids = [a + b for b in self.prefix_chars for a in self.prefix_chars]
         n = len(self.replicas)
         self.work_result = {k: [0] * n for k in self.work_result.keys()}
 
-    walker_state_spec = dict(tokens=None,
-                             row=None)
+    walker_state_spec = dict(tokens=None, row=None)
 
     def walker_initialize(self) -> None:
         n = len(self.replicas)
         self.row: Tuple[Optional[str], ...] = (None,) * n
```

Comment:
I think `self.row` gets turned into a list when it's round-tripped through Step Functions state:
```
In [8]: json.loads(json.dumps((1,2,3)))
Out[8]: [1, 2, 3]
```

Reply:
That's the intent. Do you see any problems with that?

Reply:
No, just confused by the initialization to an immutable type.

Reply:
… other than the type annotation being inaccurate? This is only one of two assignments and while the first one could be changed to a list, the second one is more or less bound to be a tuple because that's what `zip` conceptually is, an iterable of tuples. Tuples are cheaper than lists. I think I tried to fix the type hint this but the problem was that there is no type class in `typing` that reflects the support of the `[]` operator with an integer argument, i.e. `def __get__(i:int)`. I'll take another look.

The visitation code notoriously defeats the type checkers, both PyCharm and mypy, mostly because of the state (de)serialization magic. I should add this as one of my criticisms to https://github.com/HumanCellAtlas/data-store/issues/813.

Reply:
In that case, perhaps postpone assignment to `self.row` until after the `while` loop.

Reply:
Maybe I should just explicitly convert the tuple to a list in the second assignment and do the assignment outside of the loop.